### PR TITLE
Use percent slices for splits

### LINKED ISSUE #880: Specify train/val/test splits as percentages instead of n_batches or file indices
We use `cached_simulator.train_n_batches` to determine the size of the training set, and `cached_simulator.(val|test)_split_file_indices` for the validation/test sets. Instead, we should abstract this away and just specify the percent of the total data to use as each split. For example, `splits: 80/10/10` would mean use 80% of the total data for training, and 10% each for validation/test.

`CachedSimulatedDataset` already loads all available data into `self.(data|valid|test)`, so we would just change it to instead load it all into `self.data` in the constructor and slice it in `(train|val|test)_dataloader` based on the config.

### PR DESCRIPTION
**Pull request by aakashdp6548**
Fixes #880. Parsing full slices ("0:80/80:90/90:100") was easier than relative values ("80/10/10") so I just went with this for now, since it allows for more flexibility anyway. Not sure if we need more sophisticated input validation - let me know if you think I should add it.

### CODE REVIEW

**Review by jeff-regier — APPROVED** [review]
Nice! Thanks.

**Review by zhixiangteoh — APPROVED** [review]
Good stuff! Thanks!

File: bliss/simulator/simulated_dataset.py [hunk 1]
```diff
@@ -190,77 +190,74 @@ def test_dataloader(self):
...
-            if filename.startswith(self.file_prefix) and filename.endswith(".pt"):
-                self.data += self.read_file(f"{self.cached_data_path}/{filename}")
+        # parse slices from percentages to indices
+        self.slices = self.parse_slices(splits, len(self.data))
 
-        # fix validation set
-        for idx in self.val_split_file_idxs:
-            filename = f"{self.file_prefix}_{idx}.pt"
-            self.valid += self.read_file(f"{self.cached_data_path}/{filename}")
+    def pct_to_idx(self, x, length):
```
[the remark this answers is not in the document]
Comment (aakashdp6548) [reply bliss/simulator/simulated_dataset.py]: Fixed

File: bliss/simulator/simulated_dataset.py [hunk 2]
```diff
@@ -190,77 +190,74 @@ def test_dataloader(self):
...
+        return int(float(x.strip()) / 100 * length) if x.strip() else None
 
-        # fix test set
-        for idx in self.test_split_file_idxs:
-            filename = f"{self.file_prefix}_{idx}.pt"
-            self.test += self.read_file(f"{self.cached_data_path}/{filename}")
+    def parse_slices(self, splits: str, length: int):
+        slices = [slice(0, 0) for _ in range(3)]  # default to empty slice for each split
+        for i, data_split in enumerate(splits.split("/")):
+            # map "start_pct:stop_pct" to slice(start_idx, stop_idx)
```
[the remark this answers is not in the document]
Comment (aakashdp6548) [reply bliss/simulator/simulated_dataset.py]: Fixed

**Pull request merged**